### PR TITLE
Switch poe updates

### PR DESCRIPTION
**Pull request by jhughesbiot**

### CODE REVIEW

**Comment by grafnu** [comment]
Can you please squash and force-push your commit history?  Having so many commits in the PR makes it hard to review (because the web-pages lists them all out).

**Comment by grafnu** [comment]
The root problem is that "git merge" (or pull, which does a merge)
preserves a lot of history and context which over time causes cruft
buildup.  So, you either need to periodically create a new branch from
master, rebase (alternate to merge), or reset.  The easiest is reset -- I
just did it for one of my branches -- see below.  None of these commands
change any files on disk (except maybe for the initial git merge from
master), they just play with the git history.

peringknife@peringknife-glaptop3:~/daq$ *git merge master*
Already up to date.
peringknife@peringknife-glaptop3:~/daq$ *git reset master*
Unstaged changes after reset:
M bin/setup_dev
...
M testing/test_aux.sh
peringknife@peringknife-glaptop3:~/daq$ *git add .*
peringknife@peringknife-glaptop3:~/daq$ *git commit -m "Provisioning
tweaks"*
[provisioner a05906f9] Provisioning tweaks
 14 files changed, 96 insertions(+), 15 deletions(-)
peringknife@peringknife-glaptop3:~/daq$ *git push*
To github.com:grafnu/daq.git
 ! [rejected]          provisioner -> provisioner (non-fast-forward)
error: failed to push some refs to 'github.com:grafnu/daq.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
peringknife@peringknife-glaptop3:~/daq$ *git push --force*
Enumerating objects: 59, done.
Counting objects: 100% (59/59), done.
Delta compression using up to 8 threads
Compressing objects: 100% (29/29), done.
Writing objects: 100% (30/30), 3.93 KiB | 447.00 KiB/s, done.
Total 30 (delta 26), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (26/26), completed with 26 local objects.
To github.com:grafnu/daq.git
 + 11679352...a05906f9 provisioner -> provisioner (forced update)

**Comment by jhughesbiot** [comment]
Attempted with no effect to commit history.

**Comment by grafnu** [comment]
Please capture the exact sequence of commands and output that you're doing.
Also might throw in a "git log -n 1" occasionally to see what the HEAD is
at each point.

**Comment by jhughesbiot** [comment]
```
git merge master
Already up to date.

 git log -n 1
commit 78512bd5a32f9c4df9714ad258f48ee825e95fb0 (HEAD -> switch-poe-updates, origin/switch-poe-updates)
Merge: ddb0ad4 b4ad528
Author: jhughesbiot <jonathan.hughes@buildingsiot.com>
Date:   Fri Aug 28 08:57:30 2020 -0700

    Merge branch 'switch-poe-updates' of https://github.com/jhughesbiot/daq into switch-poe-updates

git reset master
Unstaged changes after reset:
M       docs/device_report.md
M       subset/switches/src/main/java/switchtest/SwitchTest.java
M       subset/switches/test_switch

git log -n 1
commit b196439b26892edfe85e21746c8f3b82a77125e4 (HEAD -> switch-poe-updates, origin/master, origin/HEAD, master)
Merge: 53acd58 861eccd
Author: jhughesbiot <jonathan.hughes@buildingsiot.com>
Date:   Fri Aug 28 08:54:01 2020 -0700

    Merge branch 'master' of https://github.com/jhughesbiot/daq

git add .
git log -n 1
commit b196439b26892edfe85e21746c8f3b82a77125e4 (HEAD -> switch-poe-updates, origin/master, origin/HEAD, master)
Merge: 53acd58 861eccd
Author: jhughesbiot <jonathan.hughes@buildingsiot.com>
Date:   Fri Aug 28 08:54:01 2020 -0700

    Merge branch 'master' of https://github.com/jhughesbiot/daq

git commit -m 'history cleanup attempt'
[switch-poe-updates 3b7f2d4] history cleanup attempt
 3 files changed, 1 insertion(+), 50 deletions(-)

git log -n 1
commit 3b7f2d47958cf820dffd7070aad996c73b18cd4c (HEAD -> switch-poe-updates)
Author: jhughesbiot <jonathan.hughes@buildingsiot.com>
Date:   Fri Aug 28 09:30:39 2020 -0700

    history cleanup attempt

git push
To https://github.com/jhughesbiot/daq
 ! [rejected]        switch-poe-updates -> switch-poe-updates (non-fast-forward)
error: failed to push some refs to 'https://github.com/jhughesbiot/daq'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.

git push --force
Username for 'https://github.com': jhughesbiot
Password for 'https://jhughesbiot@github.com':
Enumerating objects: 23, done.
Counting objects: 100% (23/23), done.
Delta compression using up to 8 threads
Compressing objects: 100% (9/9), done.
Writing objects: 100% (12/12), 876 bytes | 125.00 KiB/s, done.
Total 12 (delta 7), reused 0 (delta 0)
remote: Resolving deltas: 100% (7/7), completed with 7 local objects.
To https://github.com/jhughesbiot/daq
 + 78512bd...3b7f2d4 switch-poe-updates -> switch-poe-updates (forced update)
```

**Comment by grafnu** [comment]
Your origin/master and master are out of sync with upstream master. I'm not
sure what your origin is pointing at but that will cause confusion. I
generally keep my master branch completely in sync with upstream master, so
no local changes.  Note your master commit is a "merge commit" and is the
same as your feature branch.  The sequence to reset your master to follow
upstream faucet is something like:

peringknife@peringknife-glaptop3:~/daq$ *git remote -v | fgrep faucet*
faucet git@github.com:faucetsdn/daq.git (fetch)
faucet git@github.com:faucetsdn/daq.git (push)
peringknife@peringknife-glaptop3:~/daq$ *git checkout master*
Already on 'master'
Your branch is up to date with 'origin/master'.
peringknife@peringknife-glaptop3:~/daq$ *git fetch faucet*
peringknife@peringknife-glaptop3:~/daq$ *git reset --hard faucet/master *
HEAD is now at 55c879f3 Update dependency gradle to v6.6.1 (#617)

and then you should end up with something like:

~/daq$ *git log -n 1*
commit 55c879f350001fafbd138a800de3d4555032eaf7 (HEAD -> master,
origin/release_testing, origin/master, origin/HEAD, faucet/release_testing,
faucet/master, release_testing)
Author: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Date:   Tue Aug 25 16:24:44 2020 -0700

    Update dependency gradle to v6.6.1 (#617)

**Comment by jhughesbiot** [comment]
It was already in sync but looks like resetting the master before merging was the key.  Cleaned up now

File: docs/device_report.md [hunk 1]
```diff
@@ -75,9 +75,7 @@ Overall device result FAIL
 |skip|connection.port_link|Other|Other|No local IP has been set, check system config|
 |skip|connection.port_speed|Other|Other|No local IP has been set, check system config|
 |pass|manual.test.name|Security|Recommended|Manual test - for testing|
-|skip|poe.negotiation|Other|Other|No local IP has been set, check system config|
 |skip|poe.power|Other|Other|No local IP has been set, check system config|
```
Comment (noursaidi) [inline docs/device_report.md:78]: The test name should be `poe.switch.power`

**Comment by noursaidi** [comment]
@jhughesbiot, I've reviewed and have the following comments:
1. Test name should consistently be `poe.switch.power`
2. The poe test result is not being copied into the test report - the cause of this is seems to be the module_manifest still having the older tests. Once removed it should work.
3. The readme should be updated to remove reference to the older tests
4. The results for all tests including the POE power test says 'see log above', but there isn't any additional information.

**Comment by jhughesbiot** [comment]
@noursaidi 
I have made the necessary adjustments to the files that should generate expected results.  Due to the issue I reported a few days ago (https://github.com/faucetsdn/daq/issues/620) the switch module is in a state that can't be tested or verified at this point.  Seems something in the usi module has been broken but I do not know what.

**Comment by jhughesbiot** [comment]
@noursaidi @pisuke 
Issue previously referenced has been resolved and merged.  Looks like all tests are passing with the updates so unless there's something else we need to update, this one should be good to go.

**Comment by pisuke** [comment]
@jhughesbiot @spectre-storm @noursaidi @henry54809 @pbatta  I've just checked the module with a range of configurations and devices and there are a few things to be fixed before the test code can be merged.

1. This [PoE device report](https://firebasestorage.googleapis.com/v0/b/daq-qualification-labs.appspot.com/o/origin%2Farup-daq-lab-pri%2Frun_id%2F5f5260bb%2Ftests%2Ffinish%2Freport_0040ae10387b_2020-09-04T154355.pdf?alt=media&token=c1c70dd0-207a-4a95-9594-59eb3c8ca716) correctly states that the device is powered, but it misses to indicate the power in the summary. @jhughesbiot can you please add the power in the summary?
2. The same report shows that the output text in the switch section does not include any output. @jhughesbiot can you please add the full test output in the switch section?
3. This [non PoE device report](https://firebasestorage.googleapis.com/v0/b/daq-qualification-labs.appspot.com/o/origin%2Farup-daq-lab-pri%2Frun_id%2F5f5263d5%2Ftests%2Ffinish%2Freport_04a316ea0633_2020-09-04T155709.pdf?alt=media&token=4af75504-b6d3-4dd9-8acc-10e4cf83877a) correctly shows a skip because the device is configured as non-PoE, but the overall test summary is reported as a fail because the skip seems to be counted negatively. @henry54809 @pbatta I think that the logic in the creation of the table should omit the skips and only fail when there are true failures.
4. @jhughesbiot there is a little typo [poE](https://github.com/faucetsdn/daq/compare/master...jhughesbiot:switch-poe-updates#diff-61279cf1bdf85662e1fcb83c222383d9R108) that should become PoE. Can you please fix it?

After fixing 1, 2 and 4 we can merge the PR. Point 3 should be addressed in a separate PR.

**Comment by pisuke** [comment]
Thanks @jhughesbiot, I'll get this merged then.
@henry54809 @spectre-storm issue #620 needs to be addressed as a priority, can you raise a bug for this please? I've moved the issue on the project kanban for our tracking on github. Thanks!

**Pull request merged**